R/sort_linter.R

```diff
@@ -0,0 +1,86 @@
+#' Require usage of `sort()` over `.[order(.)]`
+#'
+#' `sort()` is the dedicated option to sort a list or vector. It is more legible
+#' and around twice as fast as `.[order(.)]`, with the gap in performance
+#' growing with the vector size.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "x[order(x)]",
+#'   linters = sort_linter()
+#' )
+#'
+#' lint(
+#'   text = "x[order(x, decreasing = TRUE)]",
+#'   linters = sort_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "x[sample(order(x))]",
+#'   linters = sort_linter()
+#' )
+#'
+#' lint(
+#'   text = "y[order(x)]",
+#'   linters = sort_linter()
+#' )
+#'
+#' @evalRd rd_tags("sort_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+sort_linter <- function() {
+  xpath <- "//OP-LEFT-BRACKET/following-sibling::expr[
```

Comment:
can we add a specific index to any of these `expr`? e.g. `following-sibling::expr[1]`?